### DISCUSSION TITLE
Arch Linux build fails since 0.27.1 with Rust 1.67.1

### DISCUSSION THREAD

**Discussion by hashworks** [vectordotdev/vector#16698] [General]
The build of the [Arch Linux vector package](https://archlinux.org/packages/community/x86_64/vector/) fails since 0.27.1. I'm not sure if it's caused by 0.27.1 or some other dependency update.

This is the error I'm getting:
```
error: internal compiler error: no errors encountered even though `delay_span_bug` issued

error: internal compiler error: broken MIR in Item(WithOptConstParam { did: DefId(0:2573 ~ vector_buffers[92e1]::variants::disk_v2::record::_::{impl#0}::resolve), const_param_did: None }) (after phase change to runtime-optimized) at bb4[53]:
                                Field projection `(*_4).field[3]` specified type `rkyv::boxed::ArchivedBox<[u8]>`, but actual type is `<rkyv::with::With<rkyv::with::With<&[u8], rkyv::with::RefAsBox>, rkyv::with::CopyOptimize> as rkyv::Archive>::Archived`
  --> lib/vector-buffers/src/variants/disk_v2/record.rs:46:10
```

[This is the full build log.](https://fb.hash.works/XkjwLcf/)

[This](https://github.com/archlinux/svntogit-community/blob/packages/vector/trunk/PKGBUILD#L40) is the build script I'm using. The `prepare()` and `build()` functions run in order. Basically it's just this:

```sh
cargo fetch --locked
cargo build \
        --frozen \
        --release \
        --locked \
        --target x86_64-unknown-linux-gnu
```

The following (build-) dependencies are provided:
* gcc-libs 12.2.1
* zlib 1.2.13
* cargo / rust 1.67.1
* protobuf 21.12
* python 3.10.9
* perl 5.36.0
* cmake 3.25.2

Any ideas?

**Comment by bruceg** [comment]
This is a known issue with rustc 1.67. I have a [draft PR](https://github.com/vectordotdev/vector/pull/16212) that is running into the same issue.

**Reply by jszwedko** [reply]
To add to this, if you want to use the same version of Rust we distribute Vector with you can use the `rust-toolchain.toml` file included in the repository.

**Reply by hashworks** [reply]
Alright, switched to rustup.
